rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,198 @@
+
+In particular, many changes have been introduced that impact compatibility outside Linux and NixOS niches. These issues can hinder development on other platforms, including but not limited to Unix-like systems.
+
+In light of this state of things, we propose a novel alternative to the current building infrastructure.
+
+We expect to accomplish, among other goals,
+
+- better code structuring and documentation;
+- improved support across platforms, especially outside NixOS and Linux
+- shorter build times;
+- improved testing;
+- more reasonable dependency management;
+- an overall improved user experience.
+
+# Detailed design
+[design]: #detailed-design
+
+A carefully crafted set of Meson files should be included in order to describe how to deploy the Nix repository, generating all the expected artifacts (command line tools, libraries, configuration files, documentation etc.)
+
+This novel building infrastructure should be able to provide at least feature parity with the current quasi-autotools implementation, albeit in a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Before the examples, some informative paragraphs:
+
+## What is Meson?
+
+Meson is an open source, multiplatform build system crafted to be fast, correct and user-friendly.
+
+According to its main site,
+
+> The main design point of Meson is that every moment a developer spends writing or debugging build definitions is a second wasted. So is every second spent waiting for the build system to actually start compiling code.
+
+Among its features, we highlight:
+
+- user-friendly non-Turing complete DSL
+  - very readable Python-esque syntax and functional, stateless semantics
+- multiplatform support
+  - among operating systems: Linux, Apple MacOS, Microsoft Windows NT
+  - among programming environments: GCC, Clang, Xcode, Visual Studio etc.
+  - among programming languages: C, C++, D, Fortran, Java, Rust etc.
+    - supports command customization
+  - cross compilation
+  - many useful modules included (pkg-config, filesystem inspection, internationalization etc.)
+- Comprehensive documentation
+  - including tutorials, reference manuals and real world projects using it
+
+## What is Ninja?
+
+Ninja is a small, speed-focused, build tool that fills a similar role of Unix `make` or GNU `gmake`.
+
+Its main feature is a low-level approach to build description. Where other build systems act like high level languages, Ninja acts like an assembly.
+
+Ninja is bare-bones and constrained by design, having only the necessary semantics to describe build dependency graphs, relegating decision-making to superior tools like Meson or CMake.
+
+Ninja DSL is human-readable, however it is not convenient to be manually written by human beings. As said before, Ninja is commonly used in tandem with other, higher-level build system.
```

Comment:
meson doesn't only generate ninja files, it can generate xcode projects, msvc projects, etc (the goooglable phrase you're looking for is meson --backend). the entire point of meson is youre buildsystem isnt tethered to a compiler or compilation environment

Reply:
Yes, of course. But that's actually supporting my suggestion which is just to make it explicit that this setup won't require writing `ninja` files manually. Because leaving aside my prior knowledge, what I understand from these paragraphs is that `meson` is a replacement for `autotools` and `ninja` for `make`, which implies that most of the build logic would be handwritten `ninja` files. Which is very misleading

Reply:
you are correct; i you want to put it into terms that resemble current tooling meson does .. "replace" autotools with the same for ninja to make. however, ninja is not the same as make. ninja is basically the assembly language of build tools, it would be like saying x86-asm and C++ are the same because you can program binaries with them